library/Rules/Email.php

```diff
@@ -1,10 +1,32 @@
 <?php
 namespace Respect\Validation\Rules;
 
+use Egulias\EmailValidator\EmailValidator;
+
 class Email extends AbstractRule
 {
+    public function __construct(EmailValidator $emailValidator = null)
+    {
+        $this->emailValidator = $emailValidator;
+    }
+
+    public function getEmailValidator()
+    {
+        if (!$this->emailValidator instanceof EmailValidator
```

Comment:
Here I think you can just ask for `null`. If not null, is already type hinted and the class exists.